Sources/Fluent/Query/Action.swift

```diff
@@ -7,4 +7,72 @@ public enum Action {
     case delete
     case create
     case modify
+    case schema(Schema)
+}
+
+public enum Schema {
+    case create([Field])
+    case modify(add: [Field], remove: [Field])
+    case delete
+}
+
+extension Action: Equatable {
+    public static func ==(lhs: Action, rhs: Action) -> Bool {
+        switch lhs {
+        case .fetch:
```

Comment:
might be worth reorganizing this switch, I find it more readable, but either way is fine:

```swift
switch (lhs, rhs) {
    case (.fetch, .fetch):
        return true
    // ...
    default:
        return false
}
```